src/main/java/org/dependencytrack/util/AnalysisCommentUtil.java

```diff
@@ -28,42 +28,85 @@ public final class AnalysisCommentUtil {
 
     private AnalysisCommentUtil() { }
 
+
+    public static void makeFirstStateComment(final QueryManager qm, final Analysis analysis, final String commenter) {
+        if (analysis.getAnalysisState() != null) {
+            addAnalysisStateComment(qm, analysis, null, analysis.getAnalysisState(), commenter);
```

Comment:
We should use `NOT_SET` here instead of `null`. `null` should not be exposed to users.

```suggestion
            addAnalysisStateComment(qm, analysis, AnalysisState.NOT_SET, analysis.getAnalysisState(), commenter);
```